src/hooks/useSendQueries.ts

```diff
@@ -0,0 +1,79 @@
+import axios from 'axios';
+import { datasetsURL, subjectsURL } from '../utils/constants';
+import {
+  QueryParams,
+  DatasetsResponse,
+  NodeOption,
+  SubjectsRequestBody,
+  SubjectsResponse,
+  SubjectsQueryParams,
+} from '../utils/types';
+
+function buildSubjectsRequestBody(
+  queryForm: QueryParams,
+  datasetSelection: string[],
+  datasetResponses: DatasetsResponse['responses'],
+  nodes: NodeOption[]
+): SubjectsRequestBody {
+  const nodeNameToUrl = new Map(nodes.map((node) => [node.NodeName, node.ApiURL]));
+  const nodeDatasets = new Map<string, string[]>();
+
+  datasetResponses
+    .filter((res) => datasetSelection.includes(res.dataset_uuid))
+    .forEach((res) => {
+      const nodeUrl = nodeNameToUrl.get(res.node_name);
+      if (!nodeUrl) {
+        return;
+      }
+
+      const datasets = nodeDatasets.get(nodeUrl) ?? [];
+      datasets.push(res.dataset_uuid);
+      nodeDatasets.set(nodeUrl, datasets);
+    });
+
+  return {
+    ...queryForm,
+    nodes: Array.from(nodeDatasets.entries()).map(([nodeUrl, datasetUuids]) => ({
+      node_url: nodeUrl,
+      dataset_uuids: datasetUuids,
+    })),
+  };
+}
+
+export default function useSendQueries(IDToken: string | undefined) {
+  async function sendDatasetsQuery(datasetsRequestBody: QueryParams) {
+    const response = await axios.post<DatasetsResponse>(datasetsURL, datasetsRequestBody, {
+      headers: {
+        ...(IDToken ? { Authorization: `Bearer ${IDToken}` } : {}),
+        'Content-Type': 'application/json',
+      },
+    });
```

Comment:
This seems a bit overly complex to me. iiuc the only reason you're using this structure here is to get a closure with the ID token. I'd say that's less clear / readable and not necessary. You could pass the token to each function separately or factor out the axios client and header - and then pass the axios instance to the functions. 

The datasets and subject query functions look good to me, and I find it much clearer to import them directly inside the components - rather than to destruct them from this parent-function.

Lastly: I don't think anything inside this file is / needs to be a hook. You don't use other hooks, don't use state, and there is no need to memoize this either. So I would suggest you move the subject and dataset query functions individually to utils.ts